<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DataPage</class>
 <widget class="QWidget" name="DataPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>数据管理</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #3c4b64;</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: white;
background-color: rgba(45, 45, 45, 180);
border-radius: 12px;
padding: 8px 15px;
letter-spacing: 2px;
font-weight: bold;
box-shadow: 0 4px 6px rgba(0, 0, 0, 60);
backdrop-filter: blur(8px);
-webkit-backdrop-filter: blur(8px);</string>
     </property>
     <property name="text">
      <string>碳  足  迹  计  算  记  录  列  表</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="searchFrame">
     <property name="styleSheet">
      <string notr="true">background-color: rgba(52, 73, 94, 120);
border-radius: 10px;
padding: 10px;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="searchLayout">
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="dateFromLabel">
        <property name="minimumSize">
         <size>
          <width>40</width>
          <height>0</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">color: white;
background-color: transparent;</string>
        </property>
        <property name="text">
         <string>从：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDateEdit" name="fromDateEdit">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QDateEdit {
  background-color: rgb(150, 150, 150);
  border-radius: 5px;
  color: white;
  padding: 2px 8px;
}
QDateEdit::drop-down {
  border: 0px;
  width: 20px;
}
QDateEdit::down-arrow {
  image: url(:/resource/calendar.png);
  width: 16px;
  height: 16px;
}</string>
        </property>
        <property name="displayFormat">
         <string>yyyy-MM-dd</string>
        </property>
        <property name="calendarPopup">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="dateToLabel">
        <property name="minimumSize">
         <size>
          <width>10</width>
          <height>0</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">color: white;
background-color: transparent;</string>
        </property>
        <property name="text">
         <string>到：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDateEdit" name="toDateEdit">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QDateEdit {
  background-color: rgb(150, 150, 150);
  border: 1px solid #34495e;
  border-radius: 5px;
  color: white;
  padding: 2px 8px;
}
QDateEdit::drop-down {
  border: 0px;
  width: 20px;
}
QDateEdit::down-arrow {
  image: url(:/resource/calendar.png);
  width: 16px;
  height: 16px;
}</string>
        </property>
        <property name="displayFormat">
         <string>yyyy-MM-dd</string>
        </property>
        <property name="calendarPopup">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="searchButton">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>30</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton {
  background-color: #3498db;
  color: white;
  border-radius: 5px;
  padding: 5px 15px;
}
QPushButton:hover {
  background-color: #2980b9;
}
QPushButton:pressed {
  background-color: #1f6dad;
}</string>
        </property>
        <property name="text">
         <string>搜索</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="resetButton">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>30</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton {
  background-color: #7f8c8d;
  color: white;
  border-radius: 5px;
  padding: 5px 15px;
}
QPushButton:hover {
  background-color: #95a5a6;
}
QPushButton:pressed {
  background-color: #5f6c6d;
}</string>
        </property>
        <property name="text">
         <string>重置</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="selectionToolsLayout">
     <item>
      <widget class="QPushButton" name="selectAllButton">
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>30</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton {
  background-color: #34495e;
  color: white;
  border-radius: 4px;
  padding: 4px 8px;
}
QPushButton:hover {
  background-color: #2c3e50;
}
QPushButton:pressed {
  background-color: #1f2c38;
}</string>
       </property>
       <property name="text">
        <string>全选/取消全选</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="selectionCountLabel">
       <property name="styleSheet">
        <string notr="true">color: white;
font-size: 12px;</string>
       </property>
       <property name="text">
        <string>已选择: 0 条记录</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="dataTableWidget">
     <property name="styleSheet">
      <string notr="true">QTableWidget {
  background-color: rgba(0, 0, 0, 130);
  alternate-background-color: rgba(44, 62, 80, 80);
  color: white;
  gridline-color: #34495e;
  border-radius: 10px;
  padding: 5px;
  selection-background-color: #2980b9;
  selection-color: white;
  outline: none;
}

QTableWidget::item {
  padding: 5px;
  border-bottom: 1px solid rgba(255, 255, 255, 20);
}

QTableWidget::item:selected {
  background-color: #3498db;
}

QHeaderView::section {
  background-color: #2c3e50;
  color: white;
  padding: 8px;
  font-weight: bold;
  border: none;
  border-right: 1px solid #34495e;
}

QTableWidget QScrollBar:vertical {
  background: #2c3e50;
  width: 10px;
  margin: 0px;
}

QTableWidget QScrollBar::handle:vertical {
  background: #34495e;
  min-height: 20px;
  border-radius: 5px;
}

QTableWidget QScrollBar::add-line:vertical,
QTableWidget QScrollBar::sub-line:vertical {
  height: 0px;
}

QTableCornerButton::section {
  background-color: #2c3e50;
  border: none;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::MultiSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="showGrid">
      <bool>false</bool>
     </property>
     <property name="gridStyle">
      <enum>Qt::DotLine</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <property name="cornerButtonEnabled">
      <bool>false</bool>
     </property>
     <attribute name="horizontalHeaderMinimumSectionSize">
      <number>50</number>
     </attribute>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>100</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <column>
      <property name="text">
       <string>ID</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>日期</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>总碳足迹(kg)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>交通</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>家庭</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>消费</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>操作</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="statsFrame">
     <property name="styleSheet">
      <string notr="true">background-color: rgba(41, 128, 185, 100);
border-radius: 10px;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="statsLayout">
      <item>
       <widget class="QLabel" name="totalRecordsLabel">
        <property name="styleSheet">
         <string notr="true">color: white;
background-color: transparent;
font-size: 14px;
font-weight: bold;</string>
        </property>
        <property name="text">
         <string>总记录数：0</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="averageLabel">
        <property name="styleSheet">
         <string notr="true">color: white;
background-color: transparent;
font-size: 14px;
font-weight: bold;</string>
        </property>
        <property name="text">
         <string>平均碳足迹：0 kg</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonsLayout">
     <item>
      <widget class="QPushButton" name="exportButton">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>40</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton {
  background-color: #27ae60;
  color: white;
  border-radius: 5px;
  padding: 5px 15px;
  font-weight: bold;
}
QPushButton:hover {
  background-color: #2ecc71;
}
QPushButton:pressed {
  background-color: #16a085;
}</string>
       </property>
       <property name="text">
        <string>导出数据</string>
       </property>
       <property name="icon">
        <iconset theme="document-save"/>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="deleteButton">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>40</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton {
  background-color: #e74c3c;
  color: white;
  border-radius: 5px;
  padding: 5px 15px;
  font-weight: bold;
}
QPushButton:hover {
  background-color: #c0392b;
}
QPushButton:pressed {
  background-color: #a52a2a;
}</string>
       </property>
       <property name="text">
        <string>删除所选记录</string>
       </property>
       <property name="icon">
        <iconset theme="edit-delete"/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
